src/services/lessons/lessons.service.js

```diff
@@ -38,53 +43,77 @@ class Lessons {
 			create: 'Failed to create the lesson.',
 			get: 'Failed to get the lesson',
 			find: 'Failed to find lessons.',
-			update: 'Failed to update the lesson.',
+			patch: 'Failed to patch the lesson.',
 			remove: 'Failed to delete the lesson.',
-		}
+			noAccess: 'You have no access.',
+		};
 	}
 
 	setup(app) {
 		this.app = app;
 	}
 
+	clearPermission(lessons) {
+		if (!Array.isArray(lessons)) {
+			lessons = [lessons];
+		}
+		lessons.forEach(l => delete l.permissions);
+		return lessons;
+	}
+
 	async find(params) {
-		const { route } = params;
+		const { route: { courseId }, user } = params;
 
 		try {
 			const lessons = await LessonModel.find({
-				courseId: route.courseId,
+				courseId,
 				deletedAt: { $exists: false },
+			}).populate({
+				path: 'permissions.group',
+				select: 'users',
 			}).select({
 				_id: 1,
 				title: 1,
 				note: 1,
 				visible: 1,
+				permissions: 1,
 				courseId: 1,
 				position: 1,
-			}).lean().exec();
+			}).lean()
+				.exec();
+
+			// todo pagination
+			const lessonsWithAccess = (lessons || []).filter(
+				l => testAccess(l.permissions, user, 'read')
+				|| testAccess(l.permissions, user, 'write'),
+			);
 
-			return lessons || [];
+
+			return this.clearPermission(lessonsWithAccess);
 		} catch (err) {
 			throw new BadRequest(this.err.find, err);
 		}
 	}
 
-	async get(id, params) {
-		const { route } = params;
+	async get(_id, params) {
+		const { route: { courseId }, user } = params;
 		let lesson;
 		const mRequest = LessonModel
 			.findOne({
-				_id: id,
-				courseId: route.courseId,
+				_id,
+				courseId,
 				deletedAt: { $exists: false },
-			}).populate('sections')
-			.select({
+			}).populate({
+				path: 'permissions.group',
+				select: 'users',
+			}).select({
 				_id: 1,
 				title: 1,
 				note: 1,
+				permissions: 1,
 				sections: 1,
 				courseId: 1,
-			});
+			}).lean();
```

Comment:
lean wird in zeile 123 hinzugefügt, falls das populate auf die sections gesetzt wird, bin ich mir nich sicher ob lean nicht erst danach definiert werden sollte

Reply:
done